examples/deploying_custom_regularizers.ipynb

```diff
@@ -0,0 +1,626 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "12016045",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Deploying Regularizers with Giotto-deep\n",
+    "\n",
+    "**Author: Henry Kirveslahti**\n",
+    "\n",
+    "In this tutorial we discuss the technical details for implementing regularizers and their use in *giotto-deep*. For a less technical introduction to regularization, please refer to the notebook *Basic Tutorial: Regularization with Giotto-deep*.\n",
+    "\n",
+    "The notebook is organized as follows:\n",
+    "\n",
+    "1. Architectural overview\n",
+    "2. Example of a custom regularizer\n",
+    "3. Hyper-parameter tuning\n",
+    "4. Regularizers that update parameters"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50af109b",
+   "metadata": {},
+   "source": [
+    "## 1. Architectural overview\n",
+    "\n",
+    "In Giotto-deep, a regularizer must have 3 methods:\n",
+    "\n",
+    "Informally, these are described as follows:\n",
+    "\n",
+    "1. The **preprocess** step contains any initialization required to set up the regularizer. This can be e.g. summaries of data that the regularizer needs.\n",
+    "2. The **update_params** step contains any complicated parameters updates that are updated once per epoch. These need not be differentiable.\n",
+    "3. The **regression_penalty**. This must be a torch-differentiable computation.\n",
+    "\n",
+    "\n",
+    "The diagram below gives an overview of these steps:\n",
+    "\n",
+    "![img](./images/reg_sw_diagram.png)\n"
```

Comment:
I think you forgot to add the picture in you PR, as I do not see it

Reply:
Right. In the end the implementation got much simplified, so I think there is no longer need for the diagram. The plain text in the notebook is changed accordingly.